### PR TITLE
Update to support PHP 8.0

### LINKED ISSUE #27: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by ocean**
Attempt fix for #27 

Signed-off-by: Drew Robinson <drew.robinson@gmail.com>

### CODE REVIEW

File: src/Reader/Reader.php [hunk 1]
```diff
@@ -315,7 +315,9 @@ public static function importString($string)
         }
 
         $libxmlErrflag = libxml_use_internal_errors(true);
-        $oldValue      = libxml_disable_entity_loader(true);
+        if (\LIBXML_VERSION < 20900) {
+            $oldValue      = libxml_disable_entity_loader(true);
+        }
```
Comment (froschdesign) [inline src/Reader/Reader.php]: Please see #25 for a better solution to avoid repetitions.

Comment (ocean) [reply src/Reader/Reader.php]: Thanks, I just saw that PR and have changed my PR to use that method switch as well 👍

**Comment by ocean** [comment]
Argh. Loads of broken tests from dependencies 😞

**Comment by froschdesign** [comment]
> Argh. Loads of broken tests from dependencies 😞

There are not so many:

* [4 errors: `setExpectedException`](https://travis-ci.com/github/laminas/laminas-feed/jobs/398620200#L457-L477) (https://phpunit.readthedocs.io/en/9.3/writing-tests-for-phpunit.html#testing-exceptions)
* [33 warnings: prophesize() is deprecated ](https://travis-ci.com/github/laminas/laminas-feed/jobs/398620200#L481-L580) (https://github.com/phpspec/prophecy-phpunit#how-to-use-it)

Both can be fixed quickly. 😃

**Comment by froschdesign** [comment]
@ocean 
Do you need help here or is something unclear with my previous comment?

**Comment by ocean** [comment]
@froschdesign No problems, I'm working through the fixes now 👍

File: src/Reader/Reader.php [hunk 2]
```diff
@@ -89,6 +89,23 @@ class Reader implements ReaderImportInterface
...
+     * 
+     * If we are using libxml >= 2.9, XML entity loading is disabled by default.
+     * 
+     * @return bool
+     */
+    public static function disableEntityLoader($flag = true)
+    {
+        if (LIBXML_VERSION < 20900) {
+            return libxml_disable_entity_loader($flag);
+        }
```
Comment (samsonasik) [inline src/Reader/Reader.php:106]: please add `return false` as fallback if `LIBXML_VERSION >= 20900`

```suggestion
        }
        
        return false;
```

Comment (froschdesign) [reply src/Reader/Reader.php:106]: @samsonasik 
With version 2.9 or higher the entity loading is disabled by default. So `false` is wrong here. Return the value of `$flag` instead.

Comment (samsonasik) [reply src/Reader/Reader.php:106]: Oh yeah, return `$flag` instead 👍

Comment (ocean) [reply src/Reader/Reader.php:106]: Ok, thanks for feedback, done.

**Comment by boesing** [comment]
Thanks, @ocean

**Comment by ocean** [comment]
No problems @boesing  👍

**Comment by brandonkelly** [comment]
Any word on when 2.13 will be released with this? PHP 8 is due to be released next week.

**Pull request merged**